src/platform/Zephyr/WiFiManager.cpp

```diff
@@ -152,6 +176,32 @@ CHIP_ERROR WiFiManager::AddNetwork(const ByteSpan & ssid, const ByteSpan & crede
     return CHIP_ERROR_INTERNAL;
 }
 
+CHIP_ERROR WiFiManager::Scan(const ByteSpan & ssid, ScanCallback callback)
+{
+    net_if * const iface = GetInterface();
+    VerifyOrReturnError(iface != nullptr, CHIP_ERROR_INTERNAL);
+
+    const device * dev = net_if_get_device(iface);
+    VerifyOrReturnError(dev != nullptr, CHIP_ERROR_INTERNAL);
+
+    const wifi_nrf_dev_ops * ops = static_cast<const wifi_nrf_dev_ops *>(dev->api);
+    VerifyOrReturnError(ops != nullptr, CHIP_ERROR_INTERNAL);
+
+    mScanCallback = callback;
+
+    // TODO: Use saner API once such exists.
+    // TODO: Take 'ssid' into account.
+    VerifyOrReturnError(ops->off_api.disp_scan(dev,
```

Comment:
FYI, I thought that in the WiFi driver the scan was implemented similar to other commands like connect, i.e. I was expecting wpa_supplicant API calls here to achieve the scan functionality. But it turns out that due some RPU memory constraints its different. An explanation from Krishna:

_"...that is a good question, reason is memory constraints, in RPU we don't have memory to store full scan results (all IEs) so we only get selected data for the scan and hence a special treatment using driver than wpa_supplicant, before connecting we do a proper scan only for that SSID"_

Reply:
I think returning results one by one is fine, but still would be good to have a proper API for that.